src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -50,10 +50,37 @@ import { deepEqualWorklet } from '@/worklets/comparisons';
 
 const REMOTE_CONFIG = getRemoteConfig();
 
+function getInitialInputValues({
+  inputAsset,
+  inputAmount,
+  outputAsset,
+  outputAmount,
+  percentageToSwap,
+  sliderXPosition,
+}: {
+  inputAsset: ExtendedAnimatedAssetWithColors | null;
+  inputAmount: string | undefined;
+  outputAsset: ExtendedAnimatedAssetWithColors | null;
+  outputAmount: string | undefined;
+  percentageToSwap: number;
+  sliderXPosition: number;
+}) {
+  if (inputAsset && inputAmount) {
+    const inputNativeValue = mulWorklet(inputAmount, inputAsset?.price?.value ?? 0);
+    return { inputAmount, inputNativeValue, outputAmount: 0, outputNativeValue: 0 };
+  }
+  if (outputAsset && outputAmount) {
+    const outputNativeValue = mulWorklet(outputAmount, outputAsset?.price?.value ?? 0);
+    return { inputAmount: 0, inputNativeValue: 0, outputAmount, outputNativeValue };
+  }
```

Comment:
the idea was to prefer `inputAmount > outputAmount > percentageToSell`
can never be true at the same time because if we have `inputAmount` output is calculated from that and vice versa 
it correctly sets the inputAmount to 50% if not especified, that happens in `getInputValuesForSliderPositionWorklet `